### PR TITLE
add example

### PR DESCRIPTION
**Pull request by John-Colvin**

### CODE REVIEW

**Comment by 9il** [comment]
What about Travis? 
`dub.sdl` would be better then `dub.json` because SDL is default now.

**Comment by ghost** [comment]
@9il Does DUB 0.9.23 understand SDL?

**Comment by 9il** [comment]
I don't know. You can run `dub init anyName` and see default file format.

**Comment by ghost** [comment]
json. I tried to change the json to sdl and dub complained about missing project description. I guess I am wondering about the transition period when some people may not be using the latest dub version. Best to stick with json for a while?

**Comment by 9il** [comment]
@John-Colvin  Can you add this for Travis? 

``` yml
script:
  dub test --compiler=${DC} && dub test examples/gemmTest --compiler=${DC}
```

@jansengb OK, lets save `json`.

**Comment by John-Colvin** [comment]
I agree. Let's stick with json for now, then change over when SDL has wider support.

**Comment by 9il** [comment]
> @John-Colvin Can you add this for Travis?
> 
> script:
>   dub test --compiler=${DC} && dub test examples/gemmTest --compiler=${DC}

fixed:

```
script:
   dub test --compiler=${DC} && dub add-path examples && dub test gemmTest --compiler=${DC}
```

ping @John-Colvin

**Comment by ghost** [comment]
I'm not sure that requiring a dub project for a simple example is the right way to go. It should be easy to install a package once on a system and let users simply import it into D programs (think rdmd scripts) without using dub. I'm thinking of the python world: single `pip install numpy`, many `import numpy`.  DS packages should be installed in standardized import paths and instructions should be added to the main page about setting up the import path (system wide and/or per user, as needed). We need to set things up for users as well developers.

**Comment by 9il** [comment]
@jansengb  dub is going to be dlang's pip.

**Comment by 9il** [comment]
plus `dub.json` files are part of examples.

File: examples/gemmTest/dub.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "name": "gemmtest",
+    "dependencies": {
+        "cblas": { "path": "../../" }
+    },
+    "sourcePath": ".",
```
Comment (9il) [inline examples/gemmTest/dub.json:6]: Please use standard `source` folder.

File: examples/gemmTest/dub.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    "targetType": "executable"
```
Comment (9il) [inline examples/gemmTest/dub.json:7]: can be removed, thought.

File: examples/gemmTest/dub.json (hunk 1 above)
Comment (9il) [inline examples/gemmTest/dub.json:6]: edit: `source/app.d` file

**Comment by ghost** [comment]
@9il But as a user it should be easy to write a test.d program with an import clbas, without having to `dub --init test` and edit dub.json to add a cblas dependency. That should be the norm.

**Comment by 9il** [comment]
@jansengb rdmd requires to rebuild _all_ dependencies every time: It is very slow.

**Comment by ghost** [comment]
@9il doesn't rdmd cache builds in /tmp? It is slow the first time, but subsequent runs are very fast. Anyway, even compiling with dmd/gdc/ldc, it would be better for users to not have to create a dub project for every little program or example they want to run. Packages should be installed in standard package directories (user or system) and these directories should be part of the import path.

**Comment by 9il** [comment]
@jansengb How would you build program with rdmd and dependencies from DlangScience that have dependencies from dub packages not maintained by DlangScience?

**Comment by 9il** [comment]
I agree that pip way is very useful, however we would not invent build system only for DlangScience. The proper way is to change dub, thought.

**Comment by ghost** [comment]
@9il All the required dub packages should be installed in the standard directories at the time of installation of the DS packages. I don't know the details, I just think that requiring end users to use dub is a small extra hurdle compared starting in Python. For widespread adoption, the hurdles have to be removed.

**Comment by 9il** [comment]
@jansengb I have created the issue for dub https://github.com/D-Programming-Language/dub/issues/651 . You may want to improve my description of the issue because my english is not well.

**Comment by ghost** [comment]
@9il Your English skills are FAR better than my D and dub skills :)

**Comment by John-Colvin** [comment]
`dub` is by far the easiest way to handle dependencies in D. Improving dub is the way forward here, or at least having a tool that integrates tightly with dub.

Installation to system directories is a pain and if people are using bare dmd they still have to provide linker flags explicitly anyway.

There are various ad-hoc solutions I can think of, but they all have big downsides like e.g. linking the entirety of DlangScience.

File: examples/gemmTest/dub.json (hunk 1 above)
Comment (9il) [inline examples/gemmTest/dub.json:6]: ping @John-Colvin

**Comment by John-Colvin** [comment]
@9il Having `source` folders for simple single-file examples seems excessive.

If I'm not doing anything much more complicated than Hello World, then anything more than a single file and some build instructions (i.e. a readme or a makefile or a dub.json) makes me think of enterprise java. Not a good impression to make on a newcomer, especially a scientist. As @jansengb points out, even just the dub.json can be a bit off-putting.

What is your reason for wanting to have the full structure?

**Comment by 9il** [comment]
Full structure is more simple and predictable to me.

**Comment by 9il** [comment]
Additionally it can be more useful with IDEs with dub support (default structure is more stable for them).

**Comment by ghost** [comment]
Okay, I retract my objection to using dub projects for simple examples, at least for the time being. Hopefully someone will pick up on [rdub](https://github.com/D-Programming-Language/dub/issues/651) :-)

**Comment by John-Colvin** [comment]
@9il just having to write any sort of build-config file is enough to stop a lot of people trying something out. If we make them think they have to arrange the layout of their projects according to the "one true way", that's even worse. Even the restriction of putting things in separate folders, one per project, will annoy people.

Compare to python, where it is completely fine to have a folder full of different scripts, library files etc. and you can just import what you need, no build-scripts, no specified folder structures. Is it good, scalable programming practice? No. Is it super convenient and does it reduce the mental load (and corresponding annoyance) for newbies? Yes.

Having a standardised layout is useful and we should definitely use it for the libraries themselves, but with examples we really need to be getting people to the "Wow, look how neat that API is" stage quickly, with absolutely minimal boilerplate.

File: examples/gemmTest/dub.json (hunk 2 above)
Comment (John-Colvin) [inline examples/gemmTest/dub.json:7]: without that, dub builds it as a library.

**Comment by 9il** [comment]
OK :)

**Pull request merged**